ge/org.osate.ge.ba/src/org/osate/ge/ba/businessobjecthandlers/DispatchConditionHandler.java

```diff
@@ -0,0 +1,104 @@
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Created, in part, with funding and support from the United States Government. (see Acknowledgments file).
+ *
+ * This program includes and/or can make use of certain third party source code, object code, documentation and other
+ * files ("Third Party Software"). The Third Party Software that is used by this program is dependent upon your system
+ * configuration. By using this program, You agree to comply with any and all relevant Third Party Software terms and
+ * conditions contained in any such Third Party Software or separate license file distributed with such Third Party
+ * Software. The parties who own the Third Party Software ("Third Party Licensors") are intended third party benefici-
+ * aries to this license with respect to the terms applicable to their Third Party Software. Third Party Software li-
+ * censes only apply to the Third Party Software and not any other portion of this program or this program as a whole.
+ */
+package org.osate.ge.ba.businessobjecthandlers;
+
+import java.util.Optional;
+
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.ui.editor.model.IXtextDocument;
+import org.osate.ba.aadlba.BehaviorTransition;
+import org.osate.ge.CanonicalBusinessObjectReference;
+import org.osate.ge.GraphicalConfiguration;
+import org.osate.ge.GraphicalConfigurationBuilder;
+import org.osate.ge.RelativeBusinessObjectReference;
+import org.osate.ge.ba.BehaviorAnnexReferenceUtil;
+import org.osate.ge.ba.util.BehaviorAnnexUtil.DispatchConditionHelper;
+import org.osate.ge.ba.util.BehaviorAnnexXtextUtil;
+import org.osate.ge.businessobjecthandling.BusinessObjectHandler;
+import org.osate.ge.businessobjecthandling.GetGraphicalConfigurationContext;
+import org.osate.ge.businessobjecthandling.GetNameContext;
+import org.osate.ge.businessobjecthandling.GetNameForDiagramContext;
+import org.osate.ge.businessobjecthandling.IsApplicableContext;
+import org.osate.ge.businessobjecthandling.ReferenceContext;
+import org.osate.ge.graphics.Graphic;
+import org.osate.ge.graphics.LabelBuilder;
+import org.osate.ge.internal.ui.xtext.AgeXtextUtil;
+
+public class DispatchConditionHandler implements BusinessObjectHandler {
+	public final static String DISPATCH_CONDITION = BehaviorAnnexReferenceUtil.TRANSITION_TYPE + ".dispatch_condition";
+	private final Graphic labelGraphic = LabelBuilder.create().build();
+
+	@Override
+	public boolean isApplicable(final IsApplicableContext ctx) {
+		return ctx.getBusinessObject(DispatchCondition.class).isPresent();
+	}
+
+	@Override
+	public CanonicalBusinessObjectReference getCanonicalReference(final ReferenceContext ctx) {
+		return null;
+	}
+
+	@Override
+	public RelativeBusinessObjectReference getRelativeReference(final ReferenceContext ctx) {
+		return new RelativeBusinessObjectReference(DISPATCH_CONDITION,
+				ctx.getBusinessObject(DispatchCondition.class).get().getKey());
```

Comment:
Use `orElseThrow()` instead of `get()`. It does the same thing and the no-argument version is available in java 11. It's clearer that it is your intention and avoids warnings with spot bugs.